### ISSUE TITLE
describe parallel splitting on our CI in monorepo-tools' docs

### ISSUE THREAD

**Issue by PetrHeinz** [shopsys/shopsys#408]
### What is happening

The `monorepo_split.sh` can take a while on large repostories. We have developed a better way of splitting for our CI, but haven't described it in the docs, even though it could be very helpful to others:

```bash
#!/bin/bash +x

# ANSI color codes
RED="\e[31m"
GREEN="\e[32m"
BLUE="\e[34m"
NC="\e[0m"

# Relatively new version of git must be installed
printf "\n${BLUE}Using $(git --version). The package shopsys/monorepo-tools was tested on 2.16.1.${NC}\n\n"

# List of packages to be split
PACKAGES="\
    framework \
    product-feed-zbozi \
    product-feed-google \
    product-feed-heureka \
    product-feed-heureka-delivery \
    plugin-interface \
    coding-standards \
    http-smoke-testing \
    form-types-bundle \
    migrations \
    monorepo-tools \
    project-base"

for PACKAGE in $PACKAGES; do
    # Preparing the variables to be used for splitting
    LOG_FILE="$WORKSPACE/split/$PACKAGE.log"
    if [ "$PACKAGE" == "project-base" ]; then
        SUBDIRECTORY=$PACKAGE
    else
        SUBDIRECTORY="packages/$PACKAGE"
    fi
    
    # Cloning the repository as bare into separate directory, so it can be split in a parallel process
    cd $WORKSPACE
    git clone --bare .git $WORKSPACE/split/$PACKAGE
    cd $WORKSPACE/split/$PACKAGE
    
    printf "${BLUE}$(date +%T) > Splitting package '$PACKAGE' from directory '$SUBDIRECTORY'...${NC}\n"
    printf "The progress will be logged into '$LOG_FILE'.\n\n"
    
    # Running the splitting processes in parallel
    $WORKSPACE/packages/monorepo-tools/rewrite_history_from.sh $SUBDIRECTORY master $(git tag) > $LOG_FILE 2>&1 &&
        printf "${GREEN}$(date +%T) > Splitting package '$PACKAGE' from directory '$SUBDIRECTORY' finished!${NC}\n" ||
        printf "${RED}$(date +%T) > Splitting package '$PACKAGE' from directory '$SUBDIRECTORY' failed!${NC}\n" &
done

wait

# Checking the status of the split repositories
EXIT_STATUS=0
printf "\n${BLUE}$(date +%T) > Splitting of all packages finished. Checking the ability to push the split repositories...${NC}\n\n"
for PACKAGE in $PACKAGES; do
    cd $WORKSPACE/split/$PACKAGE
    REMOTE="git@github.com:shopsys/$PACKAGE.git"
    
    git push --tags $REMOTE master --dry-run
    
    if [ $? -eq 0 ]; then
        printf "${GREEN}The split package '$PACKAGE' can be pushed into '$REMOTE'!${NC}\n"
    else
        printf "${RED}The split package '$PACKAGE' cannot be pushed into '$REMOTE'!${NC}\n"
        EXIT_STATUS=1
    fi
done

# Pushing all repositories at once if they are OK so they are released at one moment
if [ $EXIT_STATUS -eq 0 ]; then
    printf "\n${GREEN}$(date +%T) > All repositories can be pushed! Pushing them into their remotes now...${NC}\n\n"
    for PACKAGE in $PACKAGES; do
        cd $WORKSPACE/split/$PACKAGE
        git push --tags git@github.com:shopsys/$PACKAGE.git master
    done
else
printf "\n${RED}$(date +%T) > Repositories were not pushed into their remotes due to an error.${NC}\n\n"
fi

exit $EXIT_STATUS
```

### Expected result

Our CI setup for splitting the monorepo and its advantages is described in the docs.

*Note: this is duplicate of the US-3853 in our internal backlog.*

**Comment by rybakit** [comment]
@PetrHeinz What do you think about adding this script into `monorepo-tools`?

**Comment by PetrHeinz** [comment]
I would love that, but it may take a while to generalize it and make it both consistent with the other scripts and easy-to-use. At the moment, I'd rather add it to documentation as an example, which can be done fairly quickly. And of course, we can improve it afterwards.

**Comment by rybakit** [comment]
Just an idea for improving developer experience. In my script I check if a remote was already added. Otherwise it will add and fetch it for you:

```bash
split: repo_base_url := $(shell git config --get remote.origin.url | cut -f1 -d/)
split:
	@for name in $(call get_repos); do \
		( \
			if ! git config remote.$$name.url > /dev/null; then \
				git remote add $$name $(repo_base_url)/$$name.git && \
				git fetch $$name --no-tags; \
			fi; \
		) & \
	done; \
	wait
        ...
```
